bugsnag-android-unity/src/main/java/com/bugsnag/android/unity/UnityClient.java

```diff
@@ -10,32 +10,115 @@
 
 public class UnityClient {
 
+    static Client client;
+
     public static void init(Context androidContext, String apiKey, boolean trackSessions) {
+        if (client != null) {
+            client.disableExceptionHandler();
+        }
         Configuration config = new Configuration(apiKey);
         config.setAutoCaptureSessions(trackSessions);
-        Bugsnag.init(androidContext, config);
+        client = new Client(androidContext, config);
```

Comment:
I'm not sure that this method is thread-safe - `init` could be called concurrently multiple times from Unity, and if the `client` field hasn't been assigned, a client could be constructed twice or more.

Reply:
N.B. bugsnag-android uses the same pattern so any change should also be applied there